faucet.go

```diff
@@ -801,26 +839,17 @@ func (l *lightningFaucet) CloseAllChannels() error {
 	return nil
 }
 
-// cleanAndExpandPath expands environment variables and leading ~ in the passed
-// path, cleans the result, and returns it.
-// This function is taken from https://github.com/btcsuite/btcd
-func cleanAndExpandPath(path string) string {
-	// Expand initial ~ to OS specific home directory.
-	if strings.HasPrefix(path, "~") {
-		homeDir := filepath.Dir(lndHomeDir)
-		path = strings.Replace(path, "~", homeDir, 1)
-	}
-
-	// NOTE: The os.ExpandEnv doesn't work with Windows-style %VARIABLE%,
-	// but the variables can still be expanded via POSIX-style $VARIABLE.
-	return filepath.Clean(os.ExpandEnv(path))
-}
-
 // generateInvoice is a hybrid http.Handler that handles: the validation of the
 // generate invoice form, rendering errors to the form, and finally generating
 // invoice if all the parameters check out.
 func (l *lightningFaucet) generateInvoice(homeTemplate *template.Template,
 	homeState *homePageContext, w http.ResponseWriter, r *http.Request) {
+
+	// Disable generate invoice if user set this parameter
+	if l.disableGenerateInvoices {
+		return
```

Comment:
This (and the corresponding call in payInvoice) should return an http 403 error.